microsoft/testsuites/gpu/gpusuite.py

```diff
@@ -0,0 +1,90 @@
+# Licensed under the MIT license.
+
+from lisa import Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.features import Gpu, SerialConsole
+from lisa.testsuite import simple_requirement
+from lisa.tools import Lsmod, Reboot
+from lisa.util import LisaException
+
+
+@TestSuiteMetadata(
+    area="gpu",
+    category="functional",
+    description="""
+    This test suite runs the gpu test cases.
+    """,
+)
+class gpu(TestSuite):
+    @TestCaseMetadata(
+        description="""
+            This test case verfies kernel support for gpu.
+            TODO: validate the test params against GPU support test matrix
+                    when test matrix is available.
+        """,
+        requirement=simple_requirement(
+            supported_features=[Gpu],
+        ),
+    )
+    def validate_gpu_support(self, case_name: str, node: Node) -> None:
+        gpu_feature = node.features[Gpu]
+        gpu_feature.check_support()
+
+    @TestCaseMetadata(
+        description="""
+            This test case verfies if gpu drivers are loaded fine.
+
+            Steps:
+            1. Validate if the VM SKU is supported for GPU.
+            2. Install LIS drivers if not already installed for Fedora and its
+                derived distros. Reboot the node
+            3. Install required gpu drivers on the VM and reboot the node. Validate gpu
+                drivers can be loaded successfully.
+
+        """,
+        requirement=simple_requirement(
+            supported_features=[Gpu, SerialConsole],
+        ),
+    )
+    def validate_load_driver(self, case_name: str, node: Node) -> None:
+        gpu_feature = node.features[Gpu]
+        case_path = self._create_case_log_path(case_name)
+
+        gpu_feature.install_compute_sdk()
+        self.log.info(
+            f"{gpu_feature.gpu_vendor} sdk is installed. Will reboot to load driver."
+        )
+
+        reboot_tool = node.tools[Reboot]
+        reboot_tool.reboot_and_check_panic(case_path)
+
+        lsmod_tool = node.tools[Lsmod]
+        if not (lsmod_tool.module_exists(vendor) for vendor in gpu_feature.gpu_vendor):
+            raise LisaException(
+                f"{gpu_feature.gpu_vendor} GPU driver is not loaded after VM restart!"
+            )
+
+    @TestCaseMetadata(
+        description="""
+            This test case verifies the gpu adapter count.
+
+            Steps:
+            1. Find the expected gpu count for the node.
+            2. Validate expected and actual gpu count using lsvmbus output.
+            3. Validate expected and actual gpu count using lspci output.
+            4. Validate expected and actual gpu count using gpu vendor commands
+                 example - nvidia-smi
+        """,
+        requirement=simple_requirement(
+            supported_features=[Gpu, SerialConsole],
+        ),
+    )
+    def validate_gpu_adapter_count(self, case_name: str, node: Node) -> None:
+        assert self.validate_load_driver()
```

Comment:
It's better not to call other test cases. Other test case may do unnecessary steps, like reboot, it shouldn't be rebooted in this case, if the driver is already installed.